RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc

```diff
@@ -637,6 +637,11 @@ void MuonIdProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
      }
    }
 
+   if (arbitrateTrackerMuons_){
+     fillArbitrationInfo( outputMuons.get() );
```

Comment:
There is a protection already in place in fillArbitrationInfo. There is a check if a segment has already been arbitrated: https://github.com/drkovalskyi/cmssw/blob/muon-arbitration/RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc#L1037